File: backend/app/schemas/share.py
import uuid
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class ShareCreateRequest(BaseModel):
    # "1h" / "24h" / "7d" / "30d" / integer-seconds string / "never". Empty -> 7d.
    expires_in: str = "7d"


class ShareCreateResponse(BaseModel):
    token: str
    url: str
    expires_at: datetime | None


class ShareLinkOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    bucket: str
    object_key: str
    created_by_email: str
    created_at: datetime
    expires_at: datetime | None
    revoked: bool
    access_count: int
    last_accessed_at: datetime | None

File: frontend/src/api/share.ts
import apiClient from './client'

export interface ShareCreateResponse {
  token: string
  url: string
  expires_at: string | null
}

export interface ShareLink {
  id: string
  bucket: string
  object_key: string
  created_by_email: string
  created_at: string
  expires_at: string | null
  revoked: boolean
  access_count: number
  last_accessed_at: string | null
}

export const shareApi = {
  create: (bucket: string, key: string, expiresIn: string = '7d') =>
    apiClient.post<ShareCreateResponse>(
      `/share/${bucket}/${key}`,
      { expires_in: expiresIn },
    ),
  list: (bucket?: string) =>
    apiClient.get<ShareLink[]>('/share', { params: bucket ? { bucket } : {} }),
  revoke: (id: string) => apiClient.delete(`/share/${id}`),
}
